core/clustersmngr/factory_caches_test.go

```diff
@@ -99,6 +102,18 @@ func TestClusterSet_Set(t *testing.T) {
 		t.Fatalf("failed to calculate removed:\n%s", diff)
 	}
 
+	cw.Stop()
+
+	go func() {
+		for u := range cw.updates {
+			if diff := cmp.Diff(
```

Comment:
This should probably check for the number of messages received over the channel?